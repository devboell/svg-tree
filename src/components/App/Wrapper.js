import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-flow: column;
  align-items: center;
`

export default Wrapper
