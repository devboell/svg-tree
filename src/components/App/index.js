import React from 'react'
import SVGTree from 'SVGTree'
import { transformTree } from 'transformTree'
import data from 'data/plantagenet'
import Wrapper from './Wrapper'

const dimensions = {
  boxWidth: 80,
  boxHeight: 60,
  marginX: 30,
  marginY: 30,
}

const App = () =>
  <Wrapper>
    <SVGTree
      treeData={transformTree(data, dimensions)}
      {...{ dimensions }}
    />
  </Wrapper>

export default App
