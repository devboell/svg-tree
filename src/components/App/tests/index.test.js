import React from 'react'

import App from '../index'

test('App component', () => {
  const wrapper = render(<App />)
  expect(wrapper).toMatchSnapshot()
})
