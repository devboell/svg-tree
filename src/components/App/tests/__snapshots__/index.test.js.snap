// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<div
  class="sc-bwzfXH jTAEBk"
>
  <svg
    height="452"
    width="992"
  >
    <g
      fill="none"
      stroke="black"
      stroke-width="1"
      style="shape-rendering:crispEdges;"
    >
      <foreignobject
        height="60"
        width="80"
        x="415"
        y="0"
      >
        <div
          class="sc-bdVaJa iTJoaa"
          height="60"
          width="80"
        >
          <p>
            William I
          </p>
        </div>
      </foreignobject>
      <g />
      <g>
        <foreignobject
          height="60"
          width="80"
          x="525"
          y="0"
        >
          <div
            class="sc-bdVaJa iTJoaa"
            height="60"
            width="80"
          >
            <p>
              Matilde of Flanders
            </p>
          </div>
        </foreignobject>
        <path
          d="M 525 30 H 497"
          stroke="red"
        />
        <g
          fill="none"
          stroke="black"
          stroke-width="1"
          style="shape-rendering:crispEdges;"
        >
          <foreignobject
            height="60"
            width="80"
            x="30"
            y="90"
          >
            <div
              class="sc-bdVaJa iTJoaa"
              height="60"
              width="80"
            >
              <p>
                William Rufus
              </p>
            </div>
          </foreignobject>
          <g>
            <path
              d="M 70 90 V 75 H 510 V 30"
            />
          </g>
        </g>
        <g
          fill="none"
          stroke="black"
          stroke-width="1"
          style="shape-rendering:crispEdges;"
        >
          <foreignobject
            height="60"
            width="80"
            x="195"
            y="90"
          >
            <div
              class="sc-bdVaJa iTJoaa"
              height="60"
              width="80"
            >
              <p>
                Henry I
              </p>
            </div>
          </foreignobject>
          <g>
            <path
              d="M 235 90 V 75 H 510 V 30"
            />
          </g>
          <g>
            <foreignobject
              height="60"
              width="80"
              x="305"
              y="90"
            >
              <div
                class="sc-bdVaJa iTJoaa"
                height="60"
                width="80"
              >
                <p>
                  Matilde of Scotland
                </p>
              </div>
            </foreignobject>
            <path
              d="M 305 120 H 277"
              stroke="red"
            />
            <g
              fill="none"
              stroke="black"
              stroke-width="1"
              style="shape-rendering:crispEdges;"
            >
              <foreignobject
                height="60"
                width="80"
                x="140"
                y="180"
              >
                <div
                  class="sc-bdVaJa iTJoaa"
                  height="60"
                  width="80"
                >
                  <p>
                    William Adelin
                  </p>
                </div>
              </foreignobject>
              <g>
                <path
                  d="M 180 180 V 165 H 290 V 120"
                />
              </g>
            </g>
            <g
              fill="none"
              stroke="black"
              stroke-width="1"
              style="shape-rendering:crispEdges;"
            >
              <foreignobject
                height="60"
                width="80"
                x="250"
                y="180"
              >
                <div
                  class="sc-bdVaJa iTJoaa"
                  height="60"
                  width="80"
                >
                  <p>
                    Empress Matilde
                  </p>
                </div>
              </foreignobject>
              <g>
                <path
                  d="M 290 180 V 165 H 290 V 120"
                />
              </g>
              <g>
                <foreignobject
                  height="60"
                  width="80"
                  x="360"
                  y="180"
                >
                  <div
                    class="sc-bdVaJa iTJoaa"
                    height="60"
                    width="80"
                  >
                    <p>
                      Geoffrey Plantagenet
                    </p>
                  </div>
                </foreignobject>
                <path
                  d="M 360 210 H 332"
                  stroke="red"
                />
                <g
                  fill="none"
                  stroke="black"
                  stroke-width="1"
                  style="shape-rendering:crispEdges;"
                >
                  <foreignobject
                    height="60"
                    width="80"
                    x="250"
                    y="270"
                  >
                    <div
                      class="sc-bdVaJa iTJoaa"
                      height="60"
                      width="80"
                    >
                      <p>
                        Henry II
                      </p>
                    </div>
                  </foreignobject>
                  <g>
                    <path
                      d="M 290 270 V 255 H 345 V 210"
                    />
                  </g>
                  <g>
                    <foreignobject
                      height="60"
                      width="80"
                      x="360"
                      y="270"
                    >
                      <div
                        class="sc-bdVaJa iTJoaa"
                        height="60"
                        width="80"
                      >
                        <p>
                          Eleanor of Aquitane
                        </p>
                      </div>
                    </foreignobject>
                    <path
                      d="M 360 300 H 332"
                      stroke="red"
                    />
                    <g
                      fill="none"
                      stroke="black"
                      stroke-width="1"
                      style="shape-rendering:crispEdges;"
                    >
                      <foreignobject
                        height="60"
                        width="80"
                        x="250"
                        y="360"
                      >
                        <div
                          class="sc-bdVaJa iTJoaa"
                          height="60"
                          width="80"
                        >
                          <p>
                            Richard I
                          </p>
                        </div>
                      </foreignobject>
                      <g>
                        <path
                          d="M 290 360 V 345 H 345 V 300"
                        />
                      </g>
                    </g>
                    <g
                      fill="none"
                      stroke="black"
                      stroke-width="1"
                      style="shape-rendering:crispEdges;"
                    >
                      <foreignobject
                        height="60"
                        width="80"
                        x="360"
                        y="360"
                      >
                        <div
                          class="sc-bdVaJa iTJoaa"
                          height="60"
                          width="80"
                        >
                          <p>
                            John I
                          </p>
                        </div>
                      </foreignobject>
                      <g>
                        <path
                          d="M 400 360 V 345 H 345 V 300"
                        />
                      </g>
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
          <g>
            <foreignobject
              height="60"
              width="80"
              x="580"
              y="90"
            >
              <div
                class="sc-bdVaJa iTJoaa"
                height="60"
                width="80"
              >
                <p>
                  Adeliza of Louvain
                </p>
              </div>
            </foreignobject>
            <path
              d="M 580 120 H 387"
              stroke="red"
            />
          </g>
        </g>
        <g
          fill="none"
          stroke="black"
          stroke-width="1"
          style="shape-rendering:crispEdges;"
        >
          <foreignobject
            height="60"
            width="80"
            x="690"
            y="90"
          >
            <div
              class="sc-bdVaJa iTJoaa"
              height="60"
              width="80"
            >
              <p>
                Robert Curthose
              </p>
            </div>
          </foreignobject>
          <g>
            <path
              d="M 730 90 V 75 H 510 V 30"
            />
          </g>
        </g>
        <g
          fill="none"
          stroke="black"
          stroke-width="1"
          style="shape-rendering:crispEdges;"
        >
          <foreignobject
            height="60"
            width="80"
            x="800"
            y="90"
          >
            <div
              class="sc-bdVaJa iTJoaa"
              height="60"
              width="80"
            >
              <p>
                Adela of Normandy
              </p>
            </div>
          </foreignobject>
          <g>
            <path
              d="M 840 90 V 75 H 510 V 30"
            />
          </g>
          <g>
            <foreignobject
              height="60"
              width="80"
              x="910"
              y="90"
            >
              <div
                class="sc-bdVaJa iTJoaa"
                height="60"
                width="80"
              >
                <p>
                  Stephen II Henry
                </p>
              </div>
            </foreignobject>
            <path
              d="M 910 120 H 882"
              stroke="red"
            />
            <g
              fill="none"
              stroke="black"
              stroke-width="1"
              style="shape-rendering:crispEdges;"
            >
              <foreignobject
                height="60"
                width="80"
                x="800"
                y="180"
              >
                <div
                  class="sc-bdVaJa iTJoaa"
                  height="60"
                  width="80"
                >
                  <p>
                    Stephen of Blois
                  </p>
                </div>
              </foreignobject>
              <g>
                <path
                  d="M 840 180 V 165 H 895 V 120"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
</div>
`;
